tools/shared/src/main/scala/org/scalajs/core/tools/linker/backend/emitter/ClassEmitter.scala

```diff
@@ -133,8 +133,43 @@ private[emitter] final class ClassEmitter(jsGen: JSGen) {
       yield js.ClassDef(Some(classIdent), parentVar, members)(tree.pos)
   }
 
+  /** Extracts the inlined init constructor, if there is one. */
+  def extractInlinedInit(tree: LinkedClass)(
+      implicit globalKnowledge: GlobalKnowledge): (Option[LinkedMember[MethodDef]], List[LinkedMember[MethodDef]]) = {
+
+    val memberMethods = tree.memberMethods
+
+    if (globalKnowledge.hasInlinedInit(tree.encodedName)) {
+      val (constructors, otherMethods) = memberMethods.partition { method =>
+        Definitions.isConstructorName(method.info.encodedName)
+      }
+      assert(constructors.size == 1,
+          s"Found ${constructors.size} constructors in class " +
+          s"${tree.encodedName} which has an inlined init.")
+      (Some(constructors.head), otherMethods)
+    } else {
+      (None, memberMethods)
+    }
+  }
+
+  /** Generates an inlined init. */
+  def genInlinedInit(className: String, method: MethodDef)(
```

Comment:
I'm not sure the general purpose method `genInlinedInit` brings any value here... It's a `genMethod` essentially :S

In any case, the name or the doc needs to improve. It is absolutely not clear what parts of the overall inlined constructor are generated. (and "Generates an inlined init" as a doc comment on `genInlineInit` is frankly useless).